### PR TITLE
fix: all resources have name type param

### PR DESCRIPTION
**Pull request by thantos**

### CODE REVIEW

File: packages/@eventual/core/src/workflow.ts [hunk 1]
```diff
@@ -136,30 +141,42 @@ export interface Workflow<in Input = any, Output = any> {
...
+  Output = any
+>(
+  name: Name,
   opts: WorkflowDefinitionOptions,
   definition: WorkflowHandler<Input, Output>
-): Workflow<Input, Output>;
-export function workflow<Input = any, Output = any>(
+): Workflow<Name, Input, Output>;
+export function workflow<
+  Name extends string = string,
```
Comment (sam-goodwin) [inline packages/@eventual/core/src/workflow.ts:162]: Do we need it on workflow? Does it have a representation on the cdk side? Maybe monitoring?

Comment (thantos) [reply packages/@eventual/core/src/workflow.ts:162]: I think we should be consistent. A workflow may be addressable if it supported overrides to the destination of task, injection of env variables, permission to invoke it, or analytics.

Comment (thantos) [reply packages/@eventual/core/src/workflow.ts:162]: Same argument for transactions

**Review by sam-goodwin — APPROVED** [review]

**Review by sam-goodwin — CHANGES_REQUESTED** [review]
I don't see the changes to the way we extract the types on the cdk side? Is it still using the export name instead of the string literal? We have code to check if the export name is the same as the string name and favor it to maintain the ide connection when it's the same.

**Comment by thantos** [comment]
There were all using the types with logic to check for "same as export name" and "different from export name".

When the `Resource["name"]` resolved to a string, it is always considered to be the same as the export name. (`exportName extends string`). Now that a type parameter is used for name in the interfaces/specs, the export name is compared to the string literal type and not `string` (`exportName extends Name`).

These all work the same as commands, tasks, and subscriptions.

For:

image

Before change:

image

After change:

image

**Review by sam-goodwin — APPROVED** [review]

**Pull request merged**